cmd/nerdctl/compose.go

```diff
@@ -45,24 +46,40 @@ var composeCommand = &cli.Command{
 			Aliases: []string{"f"},
 			Usage:   "Specify an alternate compose file",
 		},
+		&cli.StringFlag{
+			Name:  "project-directory",
+			Usage: "Specify an alternate working directory",
+		},
 		&cli.StringFlag{
 			Name:    "project-name",
 			Aliases: []string{"p"},
 			Usage:   "Specify an alternate project name",
 		},
+		&cli.StringFlag{
+			Name:  "env-file",
+			Usage: "Specify an alternate environment file",
+		},
 	},
 }
 
 func getComposer(clicontext *cli.Context, client *containerd.Client) (*composer.Composer, error) {
 	nerdctlCmd, nerdctlArgs := globalFlags(clicontext)
 	o := composer.Options{
-		File:           clicontext.String("file"),
+		ProjectOptions: composecli.ProjectOptions{
+			WorkingDir:  clicontext.String("project-directory"),
+			ConfigPaths: []string{},
+			Environment: map[string]string{},
+			EnvFile:     clicontext.String("env-file"),
+		},
 		Project:        clicontext.String("project-name"),
 		NerdctlCmd:     nerdctlCmd,
 		NerdctlArgs:    nerdctlArgs,
 		DebugPrintFull: clicontext.Bool("debug-full"),
 	}
 
+	if file := clicontext.String("file"); file != "" {
+		o.ProjectOptions.ConfigPaths = append(o.ProjectOptions.ConfigPaths, file)
```

Comment:
Shouldn't this be prepended rather than appended? (in theory)

Reply:
Yep! append or prepend give the same result since you can only pass a single compose file. But when we are going to accept the passage of several files, the order is important to process an override logic. I made the change